### ISSUE TITLE
OSC52 Clipboard Provider

### ISSUE THREAD

**Issue by rumpelsepp** [neovim/neovim#14821]
tl;dr: SSH into a machine, fire up a default neovim instance -> `*` and `+` registers work out of the box!

This is a feature request for adding a builtin OSC52 clipboard provider. Currently clipboard providers are driven by several different tools which are called by neovim, e.g. `wl-copy`, `xclip`, … This approach has the drawback that it doesn't work e.g. via SSH.

There is a terminal escape sequence, [OSC52](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Operating-System-Commands), which provides clipboard support via ANSI escape codes. Once the terminal emulator supports OSC52, the clipboard could be accessed by neovim (kind of) directly without relying on external programs.

The following terminal emulators support OSC52:

* foot
* see the table here: https://github.com/ojroques/vim-oscyank

Usage:

Write `hello\n` to the clipboard:

```
$ echo -e "\e]52;c;aGVsbG8K\e\\"
```

Read from the clipboard (the terminal writes back to stdin):

```
$ echo -e "\e]52;c;?\e\\"
```

**Comment by mjlbach** [comment]
Duplicate of https://github.com/neovim/neovim/issues/3344

**Comment by rumpelsepp** [comment]
Nice, I did not find this issue because there is a space between OSC and 52 in the linked issue. What a fail. My bad.

**Comment by mjlbach** [comment]
No worries :)
